lightning/config/config.go

```diff
@@ -127,12 +127,55 @@ type Lightning struct {
 	CheckRequirements bool `toml:"check-requirements" json:"check-requirements"`
 }
 
+type PostOpLevel int
+
+const (
+	OpLevelOff PostOpLevel = iota
+	OPLevelOptional
+	OpLevelRequired
+)
+
+func (t *PostOpLevel) UnmarshalText(text []byte) error {
+	switch strings.ToLower(string(text)) {
```

Comment:
i don't think you can implement `UnmarshalText` this way. The `text` can be `"required"` or `'required'` or `"""""required"""""`. Better implement `UnmarshalTOML`.

Reply:
is there a interface `UnmarshalTOML` in the toml or encoding package? The toml souce code show table it use the `encoding.TextUnmarshaler` to decode? 
Shall we do a strings.Trim(text, `'"`) here?

Reply:
>  Shall we do a strings.Trim(text, `'"`) here?

No

> is there a interface UnmarshalTOML in the toml or encoding package? 

https://pkg.go.dev/github.com/BurntSushi/toml#example-package-UnmarshalTOML

Reply:
fixed